wallet/src/de/schildbach/wallet/ui/WalletActivity.java

```diff
@@ -300,6 +302,14 @@ public void run() {
 
         checkLowStorageAlert();
         detectUserCountry();
+        showBackupWalletDialogIfNeeded();
+    }
+
+    private void showBackupWalletDialogIfNeeded() {
+        if (showBackupWalletDialog) {
+            BackupWalletDialogFragment.show(getSupportFragmentManager());
```

Comment:
by calling BackupWalletDialogFragment.show instead of handleBackupWallet(), we skip the step of checking that the wallet is unlocked (if locked, the user enters the PIN).

Can you put handleBackupWallet() here instead?